libraries/aws/jobs/src/aws_iot_jobs_serialize.c

```diff
@@ -695,55 +694,56 @@ static AwsIotJobsError_t _generateDescribeRequest( const AwsIotJobsRequestInfo_t
     if( pJobsRequest == NULL )
     {
         IotLogError( "No memory for Jobs DESCRIBE request." );
-
-        IOT_SET_AND_GOTO_CLEANUP( AWS_IOT_JOBS_NO_MEMORY );
+        status = AWS_IOT_JOBS_NO_MEMORY;
     }
+    else
+    {
+        /* Clear the request JSON. */
+        ( void ) memset( pJobsRequest, 0x00, requestLength );
 
-    /* Clear the request JSON. */
-    ( void ) memset( pJobsRequest, 0x00, requestLength );
-
-    /* Construct the request JSON. */
-    APPEND_STRING( pJobsRequest, copyOffset, "{\"", 2 );
+        /* Construct the request JSON. */
+        APPEND_STRING( pJobsRequest, copyOffset, "{\"", 2 );
 
-    /* Add the "include job document" flag if false. */
-    if( includeJobDocument == false )
-    {
-        copyOffset = _appendFlag( pJobsRequest,
-                                  copyOffset,
-                                  INCLUDE_JOB_DOCUMENT_KEY,
-                                  INCLUDE_JOB_DOCUMENT_KEY_LENGTH,
-                                  false );
-    }
+        /* Add the "include job document" flag if false. */
+        if( includeJobDocument == false )
```

Comment:
It looks like we're appending the job document only if the caller has told us not to.

Reply:
Not sure about this. If you doubt the behavior, I can talk to the library owner.

Reply:
This block is adding just the 'include job document' flag, not the actual job document. From the comments on lines 652-653, it appears that the default value of this flag is `true`, so it does not need to be added in that case.